services/gitdiff/gitdiff.go

```diff
@@ -427,6 +427,31 @@ func (diffFile *DiffFile) ShouldBeHidden() bool {
 	return diffFile.IsGenerated || diffFile.IsViewed
 }
 
+func (diffFile *DiffFile) ModeTranslationKey(mode string) string {
+	switch mode {
+	case "040000":
+		return "git.filemode.directory"
+	case "100600":
+		return "git.filemode.normal_file_rw0_000_000"
+	case "100640":
+		return "git.filemode.normal_file_rw0_r00_000"
+	case "100644":
+		return "git.filemode.normal_file"
+	case "100700":
+		return "git.filemode.executable_file_rwx_000_000"
+	case "100750":
+		return "git.filemode.executable_file_rwx_r0x_000"
+	case "100755":
+		return "git.filemode.executable_file"
+	case "120000":
+		return "git.filemode.symbolic_link"
+	case "160000":
+		return "git.filemode.submodule"
+	default:
+		return mode
+	}
```

Comment:
Probably don't need to test for all those as there is only 5 valid for git
1.  S_IFDIR 0040000
2. S_IFLNK  0120000
3. S_IFGITLINK	0160000 (  also just happens to be S_IFDIR + S_IFLNK)
4. 0100755  (  executable ) 
5. 0100644 ( normal ) 

https://godocs.io/github.com/go-git/go-git/v5/plumbing/filemode
https://github.com/git/git/blob/master/object.h

Reply:
>  It is possible that some of the new translations will never be encountered, but it's better to be safe than sorry

Yep, I mostly added these additional ones as ChatGPT claimed they are used as well, and I didn't find any better source claiming otherwise.
Will remove the extra translations.

Reply:
interesting... the only other one I have faced was 000000 but that is a special case and its more todo with submodules that are not submodules (ie corrupted git repo)

Reply:
<details><summary>Extract from my conversation with it</summary>

> Are there any other commonly occurring filemodes in git you know of, other than 100644 and 100755?

> Yes, in Git, there are other commonly occurring file modes in addition to 100644 and 100755. The file modes in Git represent the file permission bits and are stored as a 6-digit octal number.
Here are some other commonly occurring file modes:
    040000 (also represented as 100644 in some cases): This mode is used for directories in Git. Directories are treated as files with this mode, but the content stored in them includes references to other objects (e.g., other files and subdirectories).
    160000 (also represented as 100644 in some cases): This mode is used for submodules in Git. Submodules are references to other repositories, allowing you to include external code or projects as part of your repository.
    120000: This mode is used for symbolic links in Git. Symbolic links are pointers to other files or directories.
    100640 and 100750: These modes are similar to 100644 and 100755, respectively, but with different write permissions for the group. Files with 100640 have read permissions for user and group but no execute permissions, while files with 100750 have read and execute permissions for user and group but no write permissions for group.
    100600 and 100700: These modes are used for files that should only be accessible by the user and not readable by the group or others. Files with 100600 have read and write permissions for the user and no permissions for the group and others, while files with 100700 have read, write, and execute permissions for the user and no permissions for the group and others.
These file modes are used to represent different types of files and their access permissions in a Git repository. Understanding and managing file modes is important for maintaining the integrity and security of the repository.

> How many of these can really occur?

> In Git, there are a total of 8 possible file modes that can occur. Each file mode is represented as a 6-digit octal number, and the combination of digits represents different file types and permission settings. The 8 possible file modes are:
    100644: Regular file with read and write permissions for the owner and read-only permissions for the group and others. This is the most common file mode for regular files in a Git repository.
    100755: Executable file with read and execute permissions for the owner and read and execute permissions for the group and others. This is the most common file mode for executable files in a Git repository.
    040000: Directory with read, write, and execute permissions for the owner, and read and execute permissions for the group and others. This is used for directories in Git.
    160000: Git submodule. This mode is used for submodules in Git repositories.
    120000: Symbolic link. This mode is used for symbolic links in Git repositories.
    100640: Regular file with read and write permissions for the owner and read-only permissions for the group. No execute permissions for anyone.
    100750: Regular file with read, write, and execute permissions for the owner, and read and execute permissions for the group. No permissions for others.
    100600: Regular file with read and write permissions for the owner. No permissions for the group and others.
These 8 file modes cover the various types of files and access permissions that can be encountered in a Git repository. The file modes play a crucial role in preserving the integrity and permissions of the files within the repository.

> How can I achieve the file modes `100600`, `100640`,  `100700`, or `100750` in git?

> In Git, you can achieve the desired file modes (100600, 100640, 100700, or 100750) using the git update-index command along with the --chmod option. The --chmod option allows you to change the file mode (permissions) of a file in the Git index (also known as the staging area). Note that this operation only affects the metadata in the index and does not modify the working directory or the commit history.
Here's how you can set specific file modes in Git:
    100600: To set a file with read and write permissions for the owner and no permissions for the group and others (read-only for the owner), use the following command:
```bash
git update-index --chmod=100600 path/to/your/file
```
100640: To set a file with read and write permissions for the owner and read-only permissions for the group (no permissions for others), use:
```bash
git update-index --chmod=100640 path/to/your/file
```
100700: To set a file with read, write, and execute permissions for the owner and no permissions for the group and others, use:
```bash
git update-index --chmod=100700 path/to/your/file
```
100750: To set a file with read, write, and execute permissions for the owner and read and execute permissions for the group (no permissions for others), use:
```bash
git update-index --chmod=100750 path/to/your/file
```
Please replace path/to/your/file with the actual path to the file you want to modify. After running the appropriate git update-index command, the file will have the specified file mode in the Git index.

Remember that the git update-index command only affects the Git index. If you want to apply these changes to the working directory or the repository's commit history, you need to commit the changes using git commit after setting the desired file modes.